### ISSUE TITLE
[bug] openssl version conflict, even with override=True

### ISSUE THREAD

**Issue by AtnNn** [conan-io/conan#13519]
### Environment details

* Operating System+version: Windows 11
* Compiler+version: N/A
* Conan version: 2.0.2
* Python version: 3.11

### Steps to reproduce

Given a project that requires openssl 1.1.1.t and also transitively 1.1.1s:

```
from conan import ConanFile
class Mwe(ConanFile):
  def requirements(self):
    self.requires("grpc/1.50.0#d2c60eb68fdeae68328741258897b5c8")
    self.requires("openssl/1.1.1t#7d9fc949345908e1b8609d9d9f528a08")
```

With it, I can reproduce the error from #13368. It looks like Conan can't order these version numbers.

```
$ conan lock create .
ERROR: Version conflict: grpc/1.50.0->openssl/1.1.1s, ->openssl/1.1.1t.
```

Changing the last line to

```
    self.requires("openssl/1.1.1t#7d9fc949345908e1b8609d9d9f528a08", override=True)
```

Causes `conan lock create .` to succeed, but then

```
$ conan graph info .
ERROR: Requirement 'openssl/1.1.1s' not in lockfile
```

Is this a bug? What is the proper way to depend on a different version of openssl? Thanks!

**Comment by memsharded** [comment]
Thanks very much @AtnNn for the detailed report, it helped to reproduce the issue.
It was indeed a bug in lockfiles, I am proposing #13597 to fix it, hopefully for next 2.0.3 release

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/13597 finally merged, it was more challenging than expected, it will be in next 2.0.5

**Comment by AtnNn** [comment]
Thanks! I've verified that my issue is fixed in 9dd1257e3.

**Comment by memsharded** [comment]
Fantastic, thanks for checking it @AtnNn !!
